server/src/main/java/eu/fasten/server/plugins/kafka/FastenKafkaPlugin.java

```diff
@@ -168,8 +173,12 @@ public void run() {
                     handleProducing(null, System.currentTimeMillis() / 1000L, KafkaRecordKind.NORMAL);
                 }
             }
-        } catch (Exception e) {
-            logger.error("Error occurred while processing call graphs", e);
+
+        } catch (WakeupException e) {
+            if (!closed.get()) throw e;
+
+//        } catch (Exception e) {
```

Comment:
Commented out code should always be removed (unless it serves a purpose, like documentation). Commented code is not subject to refactorings and as such will rot. The VCS history is enough to revert back to old versions...